app/endpoints/declared_endpoints.py

```diff
@@ -252,6 +258,134 @@ async def test_neuron_file(
                     L.error(f"Error deleting temporary files: {e!s}")
 
 
+# List of all available NWB Reader classes to iterate over
+
+NWB_READERS = [BBPNWBReader, ScalaNWBReader, AIBSNWBReader, TRTNWBReader]  # , VUNWBReader]
+
+
+def test_all_nwb_readers(nwb_file_path: str) -> None:
+    """Tests all registered NWB readers on the given file path.
+    Succeeds if at least one reader can successfully process the file.
+    Raises a RuntimeError if all readers fail.
+    :param nwb_file_path: The path to the NWB file.
+    :param target_protocols: The list of protocols required by the NWB readers.
+    :return: The extracted data object from the first successful reader.
+    :raises RuntimeError: If no reader is able to read the file.
+    """
+    test_protocols = ["APThreshold", "SAPThres1", "SAPThres2", "SAPThres3", "SAPTres1",
```

Comment:
For better readability, and since they are constant, I would move the list of protocols at the top level with NWB_READERS